shared/src/SystemTextJsonPayloadSerializer.cs

```diff
@@ -12,6 +14,31 @@ namespace Microsoft.Azure.Devices.Shared
     /// </summary>
     public class SystemTextJsonPayloadSerializer : PayloadSerializer
     {
+        /// <summary>
+        /// Class taken from https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?pivots=dotnet-5-0#deserialize-inferred-types-to-object-properties
+        /// </summary>
+        internal class ObjectToInferredTypesConverter : JsonConverter<object>
```

Comment:
If we need to make this visible or outside of this class I'm okay with it. However since it's purely for this implementation I thought it'd be best to just make it part of this.

Reply:
I'd vote for moving this to its own class since there is quite a bit of logic in here. If we keep it separate we can add proper comments and extend it in the future, without having to worry about it crowding our PayloadSerializer implementation.